### PR TITLE
Revert RootMeasurePolicy change #265 and align RootMeasurePolicy with Jetpack Compose

### PR DESCRIPTION
**Pull request by eymar**
This PR is about upstreaming (reverting the change we thought that need to be upstreamed).

**TL;DR;**
- Revert old [change in RootMeasurePolicy](https://github.com/JetBrains/compose-multiplatform-core/pull/265). Take the RootMeasurePolicy as it is in Jetpack Compose androidx-main
- To be discussed: Make the first (root) node take the entire window size (I did that to fix the behaviour to make it like it was in #265, but I believe we don't need to do that. More about it below).  **UPD**:  _the change caused some tests failures. I reverted it. But the discussion can continue. I left some comments below_

**Description:**
2 years ago we [changed](https://github.com/JetBrains/compose-multiplatform-core/pull/265) the RootMeasurePolicy to make the first node take the entire window size (to match Android behaviour).

Since then, our source code evolved and that change stopped behaving as it was intended to (noticed during upstreaming preparation).  But it's okay, IMO. 

Now, I believe we don't need the first (root) node to take the entire window size. Why:
- It makes it ignore the sizes set by a developer using `.size(width, height)` modifier
- If there is more than 1 node, each of them would be maxWidth, maxHeight. 

_Note_: the above 2 points are true on Android. But I don't understand [why](https://kotlinlang.slack.com/archives/G010KHY484C/p1730374648186239?thread_ts=1729697565.345709&cid=G010KHY484C)

I see only 1 reason to have the first (root) node to take the entire window size:
- As shown in https://github.com/JetBrains/compose-multiplatform-core/pull/265, it looks like the node doesn't follow LayoutDirection (RTL, LTR). 

But it was my mistake back then (not correct assumption about LocalLayoutDirection composition local). The node is layed out according its parent LayoutDirection. If we add `CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Rtl)`, all the node's children would still be layed out in Ltr. But the 2nd-level children would be layed out according to their parent LayoutDirection (Rtl).

### CODE REVIEW

File: compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/layout/RootMeasurePolicy.kt
Comment (eymar) [inline compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/layout/RootMeasurePolicy.kt:1]: The change in this file simply takes the source code from androidx-main.

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/WindowContentLayout.desktop.kt [hunk 1]
```diff
@@ -55,8 +55,13 @@ internal fun WindowContentLayout(
 
             val contentPlaceables = buildList(measurables.size) {
                 measurables.fastForEach {
-                    if (it != resizerMeasurable)
-                        add(it.measure(constraints))
+                    if (it != resizerMeasurable) {
```
Comment (eymar) [inline compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/WindowContentLayout.desktop.kt]: This makes the root node sizes take the window size. (Aligning the bahaviour with Android).
But I believe there is no strong reason to do that.

**Comment by igordmn** [comment]
Why this is a draft?

**Comment by eymar** [comment]
> Why this is a draft?

Desktop tests are failing with this change.  Investigating. 

But please review the intention. 
cc @igordmn

**Comment by igordmn** [comment]
I have no objections to make it behave as on Android. But we should be careful. If users will require to add `Modifier.fillMaxSize()`, we should mention it in Release Notes.

> If there is more than 1 node, each of them would be maxWidth, maxHeight.

There is always only one Root node. The nodes defined in `setContent` are children of this root node.

The Root node is created implicitly in `ComposeScene`.

**Comment by eymar** [comment]
> There is always only one Root node. The nodes defined in `setContent` are children of this root node.
> 
> The Root node is created implicitly in `ComposeScene`.

I mean the root node from user perspective:

```
setContent {
    Row {}
    Row {}
}
```

Two rows here. On Android both will take the max size regardless of the .size() modifiers set by a user.

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/WindowContentLayout.desktop.kt (hunk 1 above)
Comment (eymar) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/WindowContentLayout.desktop.kt]: This change made some tests fail. I reverted it in the last commit.  I think we don't really need it (the only reason to have it could be to align with Android, but again, I don't understand why it's done like that there. Asked in Slack).

Comment (eymar) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/WindowContentLayout.desktop.kt]: The desktop tests passed after this change was reverted. 
So let's continue without it.

Comment (igordmn) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/WindowContentLayout.desktop.kt]: Let's remove from the description as well.

**Review by igordmn — APPROVED** [review]
